feature_engine/creation/geo_features.py

```diff
@@ -0,0 +1,437 @@
+from feature_engine.variable_handling import check_numerical_variables
+
+# Earth's radius in different units
+EARTH_RADIUS = {
+    "km": 6371.0,
+    "miles": 3958.8,
+    "meters": 6371000.0,
+    "feet": 20902231.0,
+}
+
+
+class GeoDistanceTransformer(TransformerMixin, BaseEstimator, GetFeatureNamesOutMixin):
+    """
+    GeoDistanceTransformer() calculates the distance between two geographical
+    coordinate pairs (latitude/longitude) and adds the result as a new feature.
+
+    This transformer is useful for location-based machine learning problems such as
+    real estate pricing, delivery route optimization, ride-sharing applications,
+    and any domain where geographic proximity is relevant.
+
+    The transformer supports different distance calculation methods:
+
+    - 'haversine': Great-circle distance using the Haversine formula (default).
+      Most accurate for typical distances on Earth's surface.
+    - 'euclidean': Simple Euclidean distance in the coordinate space.
+      Fast but less accurate for long distances.
+    - 'manhattan': Manhattan (taxicab) distance in coordinate space.
+      Useful as a rough approximation for grid-based city layouts.
+
+    More details in the :ref:`User Guide <geo_distance_transformer>`.
+
+    Parameters
+    ----------
+    lat1: str
+        Column name containing the latitude of the first point.
+
+    lon1: str
+        Column name containing the longitude of the first point.
+
+    lat2: str
+        Column name containing the latitude of the second point.
+
+    lon2: str
+        Column name containing the longitude of the second point.
+
+    method: str, default='haversine'
+        The distance calculation method. Options are:
+        - 'haversine': Great-circle distance (most accurate)
+        - 'euclidean': Euclidean distance in coordinate space
+        - 'manhattan': Manhattan distance in coordinate space
+
+    output_unit: str, default='km'
+        The unit for the output distance. Options are:
+        - 'km': Kilometers
+        - 'miles': Miles
+        - 'meters': Meters
+        - 'feet': Feet
+
+    output_col: str, default='geo_distance'
+        Name of the new column containing the calculated distances.
+
+    drop_original: bool, default=False
+        Whether to drop the original coordinate columns after transformation.
+
+    Attributes
+    ----------
+    variables_:
+        List of the coordinate variables used for distance calculation.
+
+    feature_names_in_:
+        List with the names of features seen during fit.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        This transformer does not learn parameters. Validates input columns.
+
+    fit_transform:
+        Fit to data, then transform it.
+
+    transform:
+        Calculate distances and add them as a new column.
+
+    get_feature_names_out:
+        Get output feature names for transformation.
+
+    See Also
```

Comment:
I am not sure we need this section. These transformers are not related to GeoDistance's functionality.